### DISCUSSION TITLE
The field identifier must be within the allowed range

### DISCUSSION THREAD

**Discussion by Lazys** [StefanMaron/BusinessCentral.LinterCop#106] [New Rule]
Hi,
that rule is already implemented in the AppSourceCop https://docs.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/analyzers/appsourcecop-as0013.

We're not using AppSourceCop for our PerTenant extensions, but that one specific rule is really handy.

Basically I'm curious what's your approach? Are you using AppSourceCop, CodeCop, UICop and your LinterCop we should treat as sth additional to the mentioned ones or you're going to make your tool independed? I mean, are you planning to have all of the rules from other cops tools gathered in that one LinterCop?

**Comment by StefanMaron** [comment]
In general I tend to use all 4 Microsoft cops and the Linter in addition. Depending on the project I disable rules that do not fit. 

For example, for a PTE extension I disable the appsource cop rule AS0013 as I do not develop in the AppSource ID range.

But, correct me if I am wrong, I think there is already a warning in some of the MS cops if your Object ID does not go with the ID range specified in App.json

**Reply by Lazys** [reply]
For the objects IDs that's true, but not for the **fields** IDs. In my experience only AppSourceCop follows also fields IDs :)

We're going to use AppSourceCope anyway to be ready for the new licensing fee starting from 2024 for the extensions not compatible with SaaS.
